All style things. Part 1 - Config files

oswinso opened the pull request:
## Description
This PR sets up:
1. Naming conventions in `.clang-tidy`
2. Include order stuff in `.clang-format`
3. cmake styling stuff using `cmake-format` tool in `.cmake-format.py`

Note: I changed the symlink for `.clang-format` to [style-configs](https://github.com/RoboJackets/style-configs) to just the actual file.

Changes based on these config files can be seen [here](https://github.com/RoboJackets/robocup-software/pull/1568/files)

## Associated Issue
Issue #1483

oswinso committed:
clang format, cmake-format and clang tidy updates.
Also removed the symlink to style-configs.

JNeiger reviewed (approved):
No idea about the cmake stuff, everything else looks good though

kylestach commented on .clang-tidy:
```diff
@@ -28,4 +28,25 @@ Checks:          '-*,
+  - { key: readability-identifier-naming.GlobalConstantPrefix, value: "k" }
+  - { key: readability-identifier-naming.GlobalConstantCase,   value: CamelCase }
+# Functions are CamelCase.
+  - { key: readability-identifier-naming.FunctionCase,            value: CamelBack  }
```
Ideally either `CamelCase` or `lower_case` rather than `CamelBack`, right? I think I personally have a _slight_ preference towards `lower_case` but we should probably come to consensus.

oswinso replied:
lol the comment says CamelCase

oswinso replied:
If we're following google then I think it makes sense to go google all the way and do `CamelCase`.

kylestach commented on .cmake-format.py:
```diff
@@ -0,0 +1,221 @@
+# ----------------------------------
+with section("parse"):
+    # Specify structure for custom cmake functions
+    additional_commands = {'foo': {'flags': ['BAR', 'BAZ'],
```
lol?

oswinso replied:
I literally did `dump-config` and this is what it output.

oswinso replied:
Fixed.

kylestach reviewed (approved):
Approving pending the `camelBack` vs. `CamelCase` vs. `lower_case` for function names.

kylestach commented on .clang-tidy:
```diff
@@ -28,4 +28,25 @@ Checks:          '-*,
+# Variables / Parameters are lower_case.
+  - { key: readability-identifier-naming.VariableCase,            value: lower_case }
+  - { key: readability-identifier-naming.ParameterCase,           value: lower_case }
+  - { key: readability-identifier-naming.PublicMemberCase,        value: lower_case }
```
Forgot to mention, but we need a `PrivateMemberCase` and `PrivateMemberPrefix` (or suffix if that's what we decide to use).

oswinso replied:
ah that's true.

oswinso replied:
Fixed.

oswinso committed:
Fixed functions to be CamelCase.
Added PrivateMemberCase and PrivateMemberSuffix to have private members be lower_case_.

oswinso committed:
Removed the foo default additional_commands for cmake-format.py

oswinso committed:
Created scripts to run all formatting.
